MSSQL/OMOPLoader.sql

```diff
@@ -1006,11 +985,14 @@ create procedure OMOPObservationPeriod as
 begin
 
 INSERT INTO [Observation_Period]([person_id], [observation_period_start_date], [observation_period_end_date], [period_type_concept_id] ) 
-    select x.patient_num patid, case when l.patient_num is not null then l.period_start else enr_start end enr_start_date
-    , case when l.patient_num is not null then l.period_end when enr_end_end>enr_end then enr_end_end else enr_end end enr_end_date 
-    , case when l.patient_num is not null then 44814725 else 44814724 end enr_basis from 
-    (select patient_num, min(start_date) enr_start,max(start_date) enr_end,max(end_date) enr_end_end from i2b2visit where patient_num in (select person_id from person) group by patient_num) x
-    left outer join i2b2loyalty_patients l on l.patient_num=x.patient_num
+    select person_id
+		, ISNULL(min_start, ISNULL(dob, SYSDATETIME())) observation_period_start_date
+		, ISNULL(max_end, ISNULL(max_start, ISNULL(dob, SYSDATETIME()))) observation_period_end_date
+		, CASE when min_start IS NULL then 44814725 else 44814724 end period_type_concept_id
+	from (
+		select p.person_id, min(v.start_date) min_start, max(v.start_date) max_start, max(end_date) max_end, min(p.birth_datetime) dob from person p
+		left outer join i2b2visit v on p.person_id = v.patient_num

```

Comment:
Did you compare the performance of a left join to where .. in (select ...). I don't recall if I did on this one but sometimes the difference can be quite dramatic...

Reply:
Did you ever compare the performance of left join to the old where.. in (select ..)? If not let's go ahead and still merge it in but could you please add a note in the code that this changed, so if we're finding it's really slow down the road we'll remember to see if that's the problem? Thanks!

Reply:
I ran both select statements. In both cases it took only 1 second to return over 15K subjects observation periods. So no difference